streaming/src/main/scala/org/apache/spark/streaming/dstream/InputDStream.scala

```diff
@@ -47,6 +49,18 @@ abstract class InputDStream[T: ClassTag] (@transient ssc_ : StreamingContext)
   /** This is an unique identifier for the input stream. */
   val id = ssc.getNewInputStreamId()
 
+  /**
+   * A StreamingListener that maintains a maximum ingestion speed for this
+   * stream. Initialized with `attachRateEstimator`.
+   */
+  private[streaming] var rateController: Option[RateController] = None
+
+  protected[streaming] def attachRateEstimator(rateEstimator: RateEstimator): Unit = {
```

Comment:
I think this has to be public, otherwise a user cannot attach a `RateEstimator`, no?